shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/modules/surveydetail/SurveyDetailViewModel.kt

```diff
@@ -62,6 +81,40 @@ class SurveyDetailViewModel(
         }
     }
 
+    fun addAnswer(values: List<SurveySubmissionUiModel.Answer>) {
+        viewState.value.surveyDetail?.questions?.get(questionViewState.value.currentQuestionIndex)?.let {
+            val submission = SurveySubmissionUiModel(it.id, values)
+            answers.add(submission)
+        }
+        setNextQuestionState()
+    }
+
+    fun submitAnswer() {
+        val currentState = questionViewState.value
+        if(!currentState.isLoading) {
+            val newState = SurveyQuestionViewState(
+                currentState.isShowingSubmit,
+                true,
+                currentState.currentQuestionIndex
+            )
+            questionMutableViewState.update {
+                newState
+            }
+            performSubmitAnswer()
+        }
```

Comment:
Some improvement suggestions on the `currentState` declaration code style and writing parameters name for `SurveyQuestionViewState` fully 🙏
```suggestion
        with(questionViewState.value) {
            if (!isLoading) {
                questionMutableViewState.update { 
                    SurveyQuestionViewState(
                        isShowingSubmit = isShowingSubmit,
                        isLoading = true,
                        currentQuestionIndex = currentQuestionIndex
                    )
                }
                performSubmitAnswer()
            }
        }
```